governance/handler.go

```diff
@@ -32,37 +32,66 @@ import (
-	params.Ratio:                   {stringT, checkRatio, nil},
-	params.UseGiniCoeff:            {boolT, checkUint64andBool, updateUseGiniCoeff},
-	params.DeferredTxFee:           {boolT, checkUint64andBool, nil},
-	params.MinimumStake:            {stringT, checkRewardMinimumStake, nil},
-	params.StakeUpdateInterval:     {uint64T, checkUint64andBool, updateStakingUpdateInterval},
-	params.ProposerRefreshInterval: {uint64T, checkUint64andBool, updateProposerUpdateInterval},
-	params.Epoch:                   {uint64T, checkUint64andBool, nil},
-	params.Policy:                  {uint64T, checkUint64andBool, updateProposerPolicy},
-	params.CommitteeSize:           {uint64T, checkCommitteeSize, nil},
-	params.ConstTxGasHumanReadable: {uint64T, checkUint64andBool, updateTxGasHumanReadable},
-	params.Timeout:                 {uint64T, checkUint64andBool, nil},
+	params.GovernanceMode:          {isStringType, checkGovernanceMode, compareEqual, nil},
+	params.GoverningNode:           {isAddressType, checkAddressOrListOfUniqueAddresses, compareEqual, nil},
+	params.UnitPrice:               {isUint64Type, checkUint64andBool, compareEqual, updateUnitPrice},
+	params.AddValidator:            {isAddressType, checkAddressOrListOfUniqueAddresses, compareAddress, nil},
+	params.RemoveValidator:         {isAddressType, checkAddressOrListOfUniqueAddresses, compareAddress, nil},
+	params.MintingAmount:           {isStringType, checkBigInt, compareEqual, nil},
+	params.Ratio:                   {isStringType, checkRatio, compareEqual, nil},
+	params.UseGiniCoeff:            {isBoolType, checkUint64andBool, compareEqual, updateUseGiniCoeff},
+	params.DeferredTxFee:           {isBoolType, checkUint64andBool, compareEqual, nil},
+	params.MinimumStake:            {isStringType, checkRewardMinimumStake, compareEqual, nil},
+	params.StakeUpdateInterval:     {isUint64Type, checkUint64andBool, compareEqual, updateStakingUpdateInterval},
+	params.ProposerRefreshInterval: {isUint64Type, checkUint64andBool, compareEqual, updateProposerUpdateInterval},
+	params.Epoch:                   {isUint64Type, checkUint64andBool, compareEqual, nil},
+	params.Policy:                  {isUint64Type, checkUint64andBool, compareEqual, updateProposerPolicy},
+	params.CommitteeSize:           {isUint64Type, checkCommitteeSize, compareEqual, nil},
+	params.ConstTxGasHumanReadable: {isUint64Type, checkUint64andBool, compareEqual, updateTxGasHumanReadable},
+	params.Timeout:                 {isUint64Type, checkUint64andBool, compareEqual, nil},
+}
+
+func isStringType(v interface{}) bool { return stringT == reflect.TypeOf(v) }
+
+func isUint64Type(v interface{}) bool { return uint64T == reflect.TypeOf(v) }
+
+func isBoolType(v interface{}) bool { return boolT == reflect.TypeOf(v) }
+
+func isAddressType(v interface{}) bool {
+	return addressSliceT == reflect.TypeOf(v) || addressT == reflect.TypeOf(v)
+}
+
+func compareEqual(v1 interface{}, v2 interface{}) bool {
+	return v1 == v2
+}
```

Comment:
Hmm.. It seeks ok for now, but these kinds of functions can be used in various functions. What about moving theses functions into common package or similar?

Reply:
and also, it can be done like this also:

```go
func isStringType(v interface{}) bool { _, ok := v.(string); return ok }
```

Reply:
Yepp.. There was no need to implement like this. The `checkValueType` functions are moved to a `checkValue` function.
Reflected at https://github.com/klaytn/klaytn/pull/1110/commits/4caf232a66b2d7eaaa5b94a02ef1fd127745dc6e.